app/lib/account/backend.dart

```diff
@@ -317,6 +328,9 @@ class AccountBackend {
     // If the user exists, it's always cheaper to lookup the user outside a
     // transaction.
     final user = await _lookupUserByOauthUserId(auth.oauthUserId);
+    if (user != null && user.email != auth.email && auth.email != null) {
```

Comment:
I assume `auth.email != null` is a sanity check, may I suggest we add `auth.email.isNotEmpty` too? Just as a sanity check :D